doc/tikv/README.md

```diff
@@ -85,14 +85,62 @@ TODO
 
 TODO for each: why? implications, benefits
 
-* Timestamps are supplied by the client, TiKV does not get them directly from PD (exception: CDC).
-* All timestamps are unique.
-* Reads never fail (due to the transaction protocol, there might be network issues, etc. which cause failure).
-  - A locking read in a pessimistic transaction may block.
-  - A non-locking read will never block.
-* TiKV nodes do not communicate with each other, only with a client.
-* The transaction layer does not know about region topology, in particular, it does not treat regions on the same node differently to other regions.
-* If committing the primary key succeeds, then committing the secondary keys will never fail.
+### Timestamps are supplied by the client
+
+This decision benefits "user experience", performance and simplicity.
+
+First, it gives users more control over the order of concurrent transactions.
+
+For example, a client commits two transactions: T1 and then T2. 
+If timestamps are supplied by the user, it can assure that T1 won't read any effects of T2 if T1's timestamp is smaller than T2's.
+While if we let TiKV get the timestamp, the user cannot get this guarantee because the order of processing T1 and T2 is nondeterministic.
+
+Second, it simplifies the system. Otherwise we have to let TiKV maintain states of all active transactions. 
+
+Third, it is beneficial for performance. Large volume of transactions could overburden TiKV server. In addition, GC of inactive transactions is a problem.
+
+TODO: further elaboration
+
+### All timestamps are unique
+
+TODO
+
+It's true in previous versions of TiKV. Enabling 1PC or Async commit features could break this guarantee.
+
+Multiple transactions may have identical commit timestamps. Start timestamps are still unique. One transaction must have distinct start_ts and commit_ts.
```

Comment:
It's mostly correct. However, we don't define a transaction precisely. So if we take failed transactions into consideration, the start_ts and commit_ts of a failed transaction are the same 😂 (the commit_ts of a rollback record equals to its start_ts)